src/package/polyfill/createHookClass.ts

```diff
@@ -2,18 +2,50 @@ import { getMostRecentImpl } from './hmrCache';
 
 export function createHookClass(elementName: string, originalImpl: any) {
     return class extends originalImpl {
-        static get observedAttributes() {
-            return super.observedAttributes;
-        }
-
         connectedCallback() {
             const mostRecentImpl = getMostRecentImpl(elementName).prototype;
+
+            const ctx = this;
+            const observerOptions = {
+                childList: false,
+                attributes: true,
+                subtree: false
+            };
+
+            function callback(mutationList: any[], observer: MutationObserver) {
+                mutationList.forEach(mutation => {
+                    const mostRecentImpl = getMostRecentImpl(elementName).prototype;
+                    const attributes = getMostRecentImpl(elementName).__observedAttributes;
+
+                    if (
+                        mostRecentImpl.attributeChangedCallback &&
+                        attributes &&
+                        attributes.indexOf(mutation.attributeName) !== -1
+                    ) {
+                        // call back
+                        mostRecentImpl.attributeChangedCallback.apply(ctx, [
+                            mutation.attributeName,
+                            mutation.oldValue,
+                            mutation.target.getAttribute(mutation.attributeName)
+                        ]);
+                    }
+                });
+            }
+
+            // create and observe
+            (<any>this)._observer = new MutationObserver(callback);
+            (<any>this)._observer.observe((this as unknown) as Node, observerOptions);
+
             if (mostRecentImpl.connectedCallback) {
                 mostRecentImpl.connectedCallback.apply(this, arguments);
             }
         }
 
         disconnectedCallback() {
+            // cleanup
+            (<any>this)._observer.disconnect();
+            (<any>this)._observer = '';
```

Comment:
What about `delete (<any>this)._observer` to mark the reference to be GC'ed? :)

Reply:
It was supposed to be null. Dunno if delete gives us more.

Reply:
fixed (used null)